x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -61,12 +62,15 @@ public TextSimilarityRankFeaturePhaseRankCoordinatorContext(
         this.chunkScorerConfig = chunkScorerConfig;
     }
 
-    @Override
-    protected void computeScores(RankFeatureDoc[] featureDocs, ActionListener<float[]> scoreListener) {
-
-        // Wrap the provided rankListener to an ActionListener that would handle the response from the inference service
-        // and then pass the results
-        final ActionListener<InferenceAction.Response> inferenceListener = scoreListener.delegateFailureAndWrap((l, r) -> {
+    /**
+     * Creates a scoring listener that uses the resolved chunking config for proper chunking behavior.
+     */
+    private ActionListener<InferenceAction.Response> createScoringListener(
```

Comment:
This refactoring seems like it makes sense at a high level, but I think breaking out the action listener creation into a private method makes this more opaque. 

Furthermore, we need to be sensitive to the fact that chunking is not the default action, it does not happen with every request. We should only pull chunking settings from the inference endpoint if it does exist otherwise it's a wasted call.